### PR TITLE
20150718 - FrSky Sensor Hub Update

### PR DESCRIPTION
**Pull request by jihlein**
Encode arm state and flight mode into RPM, gps mode and satellite count into T1, and hdop/vdop into T2.  This drives the illustrated custom telemetry display on the Taranis.  Similar changes to the Sport Telem should also drive the telemetry display, although I can test that config.  Need to figure out how to distrubute the LUA script for those interested too.  The telemtry script was derived from one I found for DJI systems.

image

### CODE REVIEW

**Comment by peabody124** [comment]
That looks awesome! Could you link to the lua script somewhere for us to play? There is probably somewhere appropriate to check it in, perhaps the wiki though?

**Comment by jihlein** [comment]
For now, here's a temporary link to it:

https://onedrive.live.com/redir?resid=65D20090AB0ECA6E!1140&authkey=!AEndkrgW4AwGMWk&ithint=folder%2c

Both of the include folders go into the SRIPTS folder.  Rename TitanTri to the name of the model you want to use it with.  Probably should start a thread as to how it all works.....

File: flight/Modules/UAVOFrSKYSensorHubBridge/UAVOFrSKYSensorHubBridge.c [hunk 1]
```diff
@@ -400,9 +394,85 @@ static void uavoFrSKYSensorHubBridgeTask(void *parameters)
...
+			 * To work properly on Taranis, you have to set Blades to "60" in telemetry setting
+			 */
+			FlightStatusData flight_status;
+			FlightStatusGet(&flight_status);
+
+			uint16_t status = 0;
+			float hdop, vdop;
+
+			status = (flight_status.Armed == FLIGHTSTATUS_ARMED_ARMED) ? 200 : 100;
+	        status += flight_status.FlightMode;
```
Comment (peabody124) [inline flight/Modules/UAVOFrSKYSensorHubBridge/UAVOFrSKYSensorHubBridge.c]: is there a tab/spacing issue going on here or is github showing it strangely?

**Comment by peabody124** [comment]
Awesome change. Looks great to merge except for that possible indentation issue. I'm excited to port this to the the UAVOTaranis module and try it on mine.

**Comment by jihlein** [comment]
I'll check that tonight, does look odd in Github.  

I've made some changes to my multicopter Taranis program, and the LUA telem script.  Wondering if we should add a github folder somewhere for openTx artifacts relating to the TauLabs code?

**Comment by peabody124** [comment]
I would be OK with that. Possibly in shared but that is more shared things for code. Maybe an external support directory?

**Comment by jihlein** [comment]
Formatting fixed.....

**Comment by jihlein** [comment]
Found a logic error, can I add to this pull request or should I make a new branch and new pull request?

**Comment by tracernz** [comment]
Just push a new commit to this branch and it will automatically show up in this pull request.

Sent from Outlook




On Thu, Jul 23, 2015 at 4:53 PM -0700, "jihlein" <notifications@github.com> wrote:
Found a logic error, can I add to this pull request or should I make a new branch and new pull request?


---

**Comment by mlyle** [comment]
@jihlein Sorry, pushing a new commit to this branch won't work because it was already merged.

Did you submit this in another PR or does that still need to happen?

Thanks!

**Comment by jihlein** [comment]
Looks like it still needs a new PR then. I'm still on the road, I'll get too it when I get home....

**Comment by mlyle** [comment]
No rush, just wanted to make sure it didn't get lost.  -mpl

**Comment by jihlein** [comment]
Actually I must have submitted a new PR, #1744 applies the correction,

**Pull request merged**